eZ/Publish/Core/FieldType/Author/Type.php

```diff
@@ -148,4 +166,51 @@ public function isSearchable()
     {
         return true;
     }
+
+    /**
+     * Validates the fieldSettings of a FieldDefinitionCreateStruct or FieldDefinitionUpdateStruct.
+     *
+     * @param mixed $fieldSettings
+     *
+     * @return \eZ\Publish\SPI\FieldType\ValidationError[]
+     */
+    public function validateFieldSettings($fieldSettings)
+    {
+        $validationErrors = [];
+
+        foreach ($fieldSettings as $name => $value) {
+            if (!isset($this->settingsSchema[$name])) {
+                $validationErrors[] = new ValidationError(
+                    "Setting '%setting%' is unknown",
+                    null,
+                    [
+                        '%setting%' => $name,
+                    ],
+                    "[$name]"
+                );
+                continue;
+            }
+
+            switch ($name) {
+                case 'defaultAuthor':
```

Comment:
The question is whether we want to be consistent with the other implementations or improve things when it comes to FieldTypes, https://github.com/ezsystems/ezpublish-kernel/blob/master/eZ/Publish/Core/FieldType/Date/Type.php#L224.

Reply:
Each Field Type can be treated a separate, so implementations can differ. What we need to avoid is switching code style approaches between valid alternatives. For example pushing towards short array syntax was my mistake, it should be done for entire codebase at once.

In this case I find Field Type implementations quite confusing, so I would rather improve.

That said, maybe just extract contents of this `case` block to a separate method? It keeps somewhat consistency and improves readability.